### ISSUE TITLE
The experiment can't be saved if the branches names are switched 

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [mozilla/experimenter#1094]
**[Affected versions]:**
- Firefox Release 66.0 and above
- Chrome 73.0.3683.75

**[Affected Platforms]:** 	 	
- All Windows
- All Mac 	 	
- All Linux

**[Prerequisites]:**
- You are logged with your LDAP account on the "https://experimenter.stage.mozaws.net/" website.
- Have an experiment with two branches created on the Experimenter stage website.

**[Steps to reproduce]:**
1. Open the browser with the profile from prerequisites and navigate to "https://experimenter.stage.mozaws.net/".
2. Open the created experiment from prerequisites.
3. Click the "Edit" button from "Branches" section.
4. Switch the name of the branches.
5. Click the "Save Draft" button and observe the behavior.

**[Expected result]:**
- The experiment is saved.

**[Actual result]:**
- The experiment is not saved even if all input fields turn green.

**[Notes]:**
- Attached is a screen recording of the issue:
![Switch](https://user-images.githubusercontent.com/40061643/54752629-83aa1080-4be7-11e9-9a61-926aaf4df837.gif)

**Comment by jaredkerim** [comment]
This is fixed now. I'll close this, we can reopen it if it regresses.

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and is no longer reproducible on Experimenter stage "https://stage.experimenter.nonprod.dataops.mozgcp.net/" website. 
However a new behavior occurs when creating a new branch and saving the experiment with the second branch having the same name as the first one. I opened a new issue with details about this new behavior #1411.
Tested on Windows 10 x64, Mac 10.14 and Arch Linux 4.16.6 x64.
